includes/collection/class-remote-actors.php

```diff
@@ -655,9 +655,16 @@ public static function get_acct( $id ) {
 	 *
 	 * @param int $id The ID of the remote actor post.
 	 *
-	 * @return string The avatar URL or empty string if not found.
+	 * @return string The avatar URL or default avatar if not found or avatars disabled.
 	 */
 	public static function get_avatar_url( $id ) {
+		$default_avatar_url = ACTIVITYPUB_PLUGIN_URL . 'assets/img/mp.jpg';
+
+		// Respect WordPress "show avatars" setting.
+		if ( ! \get_option( 'show_avatars' ) ) {
+			return $default_avatar_url;
+		}
```

Comment:
I'm not sure we should fall back to the default avatar. WordPress' handling of the option is a matter of not displaying avatars of users interacting with the site in the front end, which is reflected in [the original question](https://sueden.social/@freuwesen/115667842788940001).

Instead, we should check all instances where we display avatars, and only do that conditionally to this option. Probably our blocks, for the most part.

Reply:
But because it was also requested: We can show a simple bar with the total numbers instead then.

Something like:

`5 Comments | 12 Likes | 3 Shares | 7 Quotes`